test/e2e/localvolume_test.go

```diff
@@ -151,6 +159,269 @@ func LocalVolumeTest(ctx *framework.Context, cleanupFuncs *[]cleanupFn) func(*te
+		}, time.Minute*5, time.Second*5).ShouldNot(gomega.HaveOccurred(), "deleting LocalVolume %q", localVolume.Name)
+
+		// verify finalizer not removed with while bound pvs exists
+		matcher.Consistently(func() bool {
+			t.Logf("verifying finalizer still exists")
+			err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: localVolume.Name, Namespace: f.Namespace}, localVolume)
+			if err != nil && (errors.IsGone(err) || errors.IsGone(err)) {
+				t.Fatalf("LocalVolume deleted with bound PVs")
+				return false
+			} else if err != nil {
+				t.Logf("error getting LocalVolume: %+v", err)
+				return false
+			}
+			return len(localVolume.ObjectMeta.Finalizers) > 0
+		}, time.Second*30, time.Second*5).Should(gomega.BeTrue(), "checking finalizer exists with bound PVs")
+		// release PV
+		t.Logf("releasing pvs")
+		eventuallyDelete(t, consumingObjectList...)
+		// verify localVolume deletion
+		matcher.Eventually(func() bool {
+			t.Log("verifying LocalVolume deletion")
+			err := f.Client.Get(goctx.TODO(), types.NamespacedName{Name: localVolume.Name, Namespace: f.Namespace}, localVolume)
+			if err != nil && (errors.IsGone(err) || errors.IsNotFound(err)) {
+				t.Logf("LocalVolume deleted: %+v", err)
+				return true
+			} else if err != nil {
+				t.Logf("error getting LocalVolume: %+v", err)
+				return false
+			}
+			t.Logf("LocalVolume found: %q with finalizers: %+v", localVolume.Name, localVolume.ObjectMeta.Finalizers)
+			return false
+		}).Should(gomega.BeTrue(), "verifying LocalVolume has been deleted", localVolume.Name)
+	}
+
+}
+
+func consumePV(t *testing.T, ctx *framework.Context, pv corev1.PersistentVolume) (*corev1.PersistentVolumeClaim, *batchv1.Job, *corev1.Pod) {
+	matcher := gomega.NewWithT(t)
+	f := framework.Global
+	name := fmt.Sprintf("%s-consumer", pv.ObjectMeta.Name)
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: f.Namespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			VolumeMode:       pv.Spec.VolumeMode,
+			StorageClassName: &pv.Spec.StorageClassName,
+			AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: pv.Spec.Capacity[corev1.ResourceStorage],
+				},
+			},
+		},
+	}
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: f.Namespace,
+			Labels: map[string]string{
+				"app":     pvConsumerLabel,
+				"pv-name": pv.Name,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app":     pvConsumerLabel,
+						"pv-name": pv.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:  "busybox",
+							Image: "busybox",
```

Comment:
busybox from docker hub cannot be pulled because of their pull limits, use `gcr.io/google_containers/busybox` or something on quay instead

Reply:
Replace with the image

Reply:
Missing `:latest`?

Reply:
No, you updated  `Name`, not `Image` :-D